src/main/cms/cms.c

```diff
@@ -307,6 +308,25 @@ static void cmsFormatFloat(int32_t value, char *floatString)
     }
 }
 
+static void cmsFormatUint8Scaled(uint8_t value, int8_t exponent, char *outputString)
+{
+    const bool fractional = (exponent < 0);
+    const int exp = ABS(constrain(exponent, -3, 2));
+
+    int multiplier = 1;
+    for (int i = 0; i < exp; i++) {
+        multiplier *= 10;
+    }
+
+    if (fractional) {
```

Comment:
I think we really should be looking into switching to https://github.com/mpaland/printf - this should make at least the printing part of this a lot simpler.

Reply:
I don't disagree, but not sure if it would have made much difference here - at least for code size. The complication in this code is that we need different scaling/decimal behavior based on the exponent. So it probably would have taken a `switch` on the exponent and define different formatting strings anyway.